dynawo/sources/Models/CPP/ModelNetwork/DYNDerivative.cpp

```diff
@@ -27,12 +27,33 @@ namespace DYN {
 
 void
 Derivatives::reset() {
-  values_.clear();
+  for (auto& value : values_) {
+    value = 0.;
+  }
+}
+
+void
+Derivatives::init() {
+  values_.reserve(50);
+  indices_.reserve(50);
 }
 
 void
 Derivatives::addValue(const int numVar, const double value) {
-  values_[numVar] += value;
+  auto it = std::find(indices_.begin(), indices_.end(), numVar);
```

Comment:
maybe we could use an additional unordered_map to register numVar->index in vector so that this find in O(N) becomes O(1)?